### ISSUE TITLE
Multiple arguments?

### ISSUE THREAD

**Issue by drcmda** [Andarist/babel-plugin-jsx-adopt#9]
Is this possible? I couldn't make it work

```jsx
<Grid >
    {(item, opened, toggle) => {
```

a possible solution could be

```jsx
const [item, opened, toggle] = adopt(<Grid />)
```

**Comment by Andarist** [comment]
That's not quite possible, because it would be ambigous with:
```js
<Grid >
    {([item, opened, toggle]) => {
```
And I can't statically know which one do you want. Supporting the given by me example allows for a good interoperability with existing, built in mechanism of context (which always provide a single argument to ur render props, which might be an array)

**Comment by drcmda** [comment]
Makes sense. adoptObj/adoptArray would work, but it's ugly. I'll carry on trying to avoid `...args`.

**Comment by Andarist** [comment]
`adaptObj` actually would not be needed because this:
```js
const { item, opened, toggle } = adopt(<Grid />)
```
is very much supported - it's a single destructured argument.

I would generally advise of avoiding `...args` in render props and stick to a single argument to keep "compatibility" with what kind of APIs React provides.
